api/bus.go

```diff
@@ -618,4 +618,29 @@ type (
 		Reporter string
 		Duration time.Duration
 	}
+
+	ContractMetric struct {
+		Time time.Time `json:"time"`
+
+		FCID types.FileContractID `json:"fcid"`
+		Host types.PublicKey      `json:"host"`
+
+		RemainingCollateral types.Currency `json:"remainingCollateral"`
+		RemainingFunds      types.Currency `json:"remainingFunds"`
+		RevisionNumber      uint64         `json:"revisionNumber"`
+
+		UploadSpending      types.Currency `json:"uploadSpending"`
+		DownloadSpending    types.Currency `json:"downloadSpending"`
+		FundAccountSpending types.Currency `json:"fundAccountSpending"`
+		DeleteSpending      types.Currency `json:"deleteSpending"`
+		ListSpending        types.Currency `json:"listSpending"`
+	}
+
+	ContractMetricsQueryOpts struct {
```

Comment:
does we need json tags here?

Reply:
I don't think so. They should be passed via query params. But if not, I'll add the tags that marshals them as json.